models/RwdSet.php

```diff
@@ -41,25 +48,26 @@ public function __construct( $key, $params ) {
 			throw new \Exception( "RwdSet::__construct() : Main image size is missing for '$key'" );
 		}
 
-		$this->key = $key;
+		$this->key = $this->clean_key( $key );
+		$this->parse_retina_options( $key );
 
 		// this means we doesn't have any responsive options (for example this is a small thumbnail).
 		if ( 1 === count( $params ) ) {
-			$this->size = new ImageSize( $key, array_shift( $params ) );
-			$this->options[$key] = new RwdOption( $key, array(
+			$this->size = new ImageSize( $this->key, array_shift( $params ), $this->retina );
+			$this->options[$this->key] = new RwdOption( $this->key, array(
 				array( $this->size->w, $this->size->h, $this->size->crop ),
 				'picture' => '<img srcset="{src}" alt="{alt}">',
 				'bg' => '',
 				'srcset' => '{w}w',
 				'sizes' => '{w}px',
-			));
+			), $this->retina);
 		} else {
-			$this->parse_options( $params );
+			$this->parse_options( $params, $this->retina );
```

Comment:
no need to pass parameter for retina. we're inside class and have $this->retina in all methods